src/historyMenuElements.ts

```diff
@@ -22,6 +23,117 @@ Gio._promisify(Gio.File.prototype, 'replace_contents_bytes_async', 'replace_cont
 // FIXME: Generated static class code produces a no-unused-expressions rule error
 /* eslint-disable no-unused-expressions */
 
+/**
+ * Preview widget at the top of the panel menu.
+ */
+class PreviewWidget extends St.Bin {
+    static [GObject.GTypeName] = 'PreviewWidget';
+
+    static {
+        GObject.registerClass(this);
+    }
+
+    private readonly previewWidth: number;
+    private readonly previewHeight: number;
+
+    /**
+     * Create a PreviewWidget
+     *
+     * @param {number} width Width of the loaded preview image.
+     */
+    constructor(width: number) {
+        let aspect;
+        // @ts-expect-error Figure out where the 'Main' variable can be imported from
```

Comment:
In the types folder is `ui/main.d.ts` already present if you want to be precise with the typing until it's properly importable.

Reply:
Changed the wording a bit and removed the null check from `Main?.`.